Automattic/wp-calypso
# JP Onboarding: Decouple Settings Network Request from Redux State Update

Introduce a new `updateJetpackOnboardingSettings` action and corresponding `JETPACK_ONBOARDING_SETTINGS_UPDATE` action type to which the JPO settings reducer will listen, thus decoupling the network `POST` request action (`saveJetpackOnboardingSettings`/`JETPACK_ONBOARDING_SETTINGS_SAVE`) from it.

To test:
1. Checkout this branch on your local Calypso.
1. Make sure your JP sandbox is running the latest Jetpack master.
1. Go to https://YourJetpackSandbox.com/wp-admin/admin.php?page=jetpack&action=onboard&calypso_env=development where `YourJetpackSandbox.com` is the domain of your Jetpack sandbox.
1. Enter a new site title and description, confirm.
1. Verify that no error is thrown.
1. Check `state.jetpackOnboarding.settings` in your browser console, verify that the new `siteTitle` and `siteDescription` are present.
1. Check the Redux log, verify that the following actions were dispatched (in this order):
   * `WPCOM_HTTP_REQUEST` (`POST` to the JP proxy endpoint, with a `path: /jetpack/v4/settings/` in `body`)
   * `JETPACK_ONBOARDING_SETTINGS_SAVE`
   * `JETPACK_ONBOARDING_SETTINGS_UPDATE`
1. Check that your JP site's title and tagline have actually been changed!
